### PR TITLE
MM-19242 - Reporting > Site Statistics: Line graphs show different dates

### PR DESCRIPTION
**Pull request by reflog**
#### Summary
Synchronize date ranges on graphs

#### Ticket Link
  Fixes https://mattermost.atlassian.net/browse/MM-19242

### CODE REVIEW

File: components/analytics/format.tsx [hunk 1]
```diff
@@ -60,6 +61,27 @@ export function formatPostsPerDayData(data: any) {
     return chartData;
 }
 
+export function syncronizeChartData(...chartDatas: any[]) {
```
Comment (lieut-data) [inline components/analytics/format.tsx]: Perhaps a high-level summary of this function for those not familiar with the data model?
```
// syncronizeChartData converges on a uniform set of labels for all entries in the given chart data. If a given label wasn't already present in the chart data, a 0-valued data point at that label is added.
//
// For date-labelled charts, this ensures that each charts starts and ends on the same interval, even if data for part of that interval was never collected.

File: components/analytics/system_analytics/system_analytics.jsx [hunk 2]
```diff
@@ -49,6 +50,7 @@ export default class SystemAnalytics extends React.PureComponent {
         const postCountsDay = formatPostsPerDayData(stats[StatTypes.POST_PER_DAY]);
         const botPostCountsDay = formatPostsPerDayData(stats[StatTypes.BOT_POST_PER_DAY]);
         const userCountsWithPostsDay = formatUsersWithPostsPerDayData(stats[StatTypes.USERS_WITH_POSTS_PER_DAY]);
+        syncronizeChartData(postCountsDay, botPostCountsDay, userCountsWithPostsDay);
```
Comment (lieut-data) [inline components/analytics/system_analytics/system_analytics.jsx]: Also, 0/5 but should this be called `synchronizeChartData`?

File: components/analytics/format.tsx (hunk 1 above)
Comment (reflog) [reply components/analytics/format.tsx]: Great suggestion. Added!

**Comment by reflog** [comment]
@lindy65 - test added

**Comment by lindy65** [comment]
Thanks @reflog!

**Pull request merged**